src/ck-ldb/lbdb.h

```diff
@@ -131,6 +131,8 @@ class LBObjUserData {
     if (d.data != NULL) {
       init();
       memcpy(data, d.data, CkpvAccess(lbobjdatalayout).size());
+    } else {
+      data = NULL;
```

Comment:
Should this also be done for the copy constructor below?

Reply:
You mean the assignment operator? It's not needed there because the operator should imply that `this` has already been initialized.

Reply:
I mean, if the (copied) `d.data == NULL`, `this->data` isn't updated. Is this intentional (ie, shouldn't it also be NULL)?

Reply:
I agree with @matthiasdiener, it seems like if `d.data` is `NULL`, then that should be propagated by `operator=`.

Reply:
I'm not sure what should happen if `d.data` is `NULL`. I'm not familiar with this class. I don't even know if the = operator is used in practice or under what circumstances. A LBObjUserData which has been init'ed (via init call) dynamically allocates an array with garbage (for user to populate later). If you are assigning a LBObjUserData with data == NULL but this.data != NULL, presumably you would want to destroy the array? I'm not sure if this situation would even happen in practice. There could also be a danger of exposing other bugs if you destroy the array without destroying the object. The user might already have a pointer to it.

Reply:
If assigning when data == NULL is not valid, then it should at least error I'd imagine. Either this same check should appear in the operator= or it should more specifically deal with the possibility, not ignore it. If there is a copy ctor, there should be an equivalent operator=, and if they don't behave the same, there should be a good reason for it.

Reply:
As a user, I'd expect the semantics of `operator=` to produce an lhs where every field matches the rhs. I'd expect the array to be destroyed if you are assigning an object with `NULL` `data`, I think.

Reply:
Or maybe the semantics of `operator=` in LBObjUserData when d.data is NULL are that it is an error and it should abort. But I'll make the change anyway since it seems most people agree on this.

Reply:
It sounds like the usage of this class is not well documented, if its not clear what is allowed, so maybe that's something that should be looked at at some point.